src/pam_google_authenticator.c

```diff
@@ -1561,6 +1561,8 @@ static int google_authenticator(pam_handle_t *pamh, int flags,
 
   // Read and process status file, then ask the user for the verification code.
   int early_updated = 0, updated = 0;
+
+  long hotp_counter = 0;
```

Comment:
if a `buf==NULL` check is added to `get_hotp_counter()` then this hotp_counter can remain const, and not be inside the below conditional.